docs/source-2.0/guides/building-models/build-config.rst

```diff
@@ -472,6 +472,52 @@ Applies the transforms defined in the given projection names.
         }
     }
 
+.. _changeStringEnumsToEnumShapes:
+
+changeStringEnumsToEnumShapes
+-----------------------------
+
+Changes string shapes with enum traits into enum shapes.
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 20 70
+
+    * - Property
+      - Type
+      - Description
+    * - synthesizeNames
+      - ``boolean``
+      - Whether enums without names should have names synthesized if possible.
+
+This transformer will attempt to convert strings with :ref:`enum traits <enum-trait>`
+into enum shapes. To successfully convert to an enum shape, each enum
+definitions within the source enum trait must have a ``name`` property, or the
+`synthesizeNames` transform config option must be enabled. By default, `synthesizeNames`
+is disabled. Additionally, if an enum definition from the enum trait is marked
+as deprecated, the :ref:`deprecated trait <deprecated-trait>` will be applied
+to the resulting enum shape member. If the enum definition is tagged as
```

Comment:
Before talking about the special handling for "internal" tag, mention that all tags from enum definition becomes `@tags` trait on the member.